### DISCUSSION TITLE
Allows ads.txt file to be added to app router root

### DISCUSSION THREAD

**Discussion by elisechant** [vercel/next.js#71633] [Ideas]
### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/dazzling-http-pdf5pv?file=%2Fapp%2Fads.txt%3A1%2C12

### To Reproduce

1. Add ads.txt to app/ (app router directory)
2. Attempt to open the file in the browser, https://pdf5pv-3000.csb.app/ads.txt 

### Current vs. Expected behavior

Expected to see ads.txt rendered, however a 404 is shown.

### Provide environment information

```bash
N/A
```

### Which area(s) are affected? (Select all that apply)

Metadata

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local), next start (local), Vercel (Deployed)

### Additional context

An installation method of Google AdSense requires adding ads.txt to root of site, similarly to adding robots.txt, Next doesnot support this https://nextjs.org/docs/app/api-reference/file-conventions/metadata. 

This can be worked around by adding the file at public and using a redirect in next.config:

```
      {
        source: '/public/ads.txt',
        destination: '/ads.txt',
        permanent: true,
      },
```

**Comment by jln-dk** [comment]
You can just add the file in the `public/` folder in the root of your project.

**Comment by elisechant** [comment]
Apologies for the run around, confirming that is correct. 

The confusion is the differing expectation between how robots.txt is handled by adding at apps/robots.txt vs adding ads.txt to public/ https://nextjs.org/docs/app/api-reference/file-conventions/metadata/robots

![Screenshot 2024-10-22 at 3 54 08 pm](https://github.com/user-attachments/assets/4e46f9be-d258-4f3a-a2c5-bbe42c8ced0c)

It would be better if there was one way of doing things.

**Comment by crossterra-agent** [comment]
- **Status**: DONE
- **Role**: Orchestrator
- **Link**: https://github.com/CrossTerra/monorepo/pull/3020#issuecomment-4624952960
- **Summary**: Reviewer marked PR #3020 COMPLETE — all five plan steps verified, CI green, no active Bugbot comments; PR ready for human review: https://github.com/CrossTerra/monorepo/pull/3020
